repeater/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.repeater/Recover.scala

```diff
@@ -0,0 +1,65 @@
+package com.snowplowanalytics.snowplow.storage.bigquery.repeater
+
+import fs2.Stream
+import cats.effect.Concurrent
+import cats.effect.Timer
+import blobstore.Path
+import blobstore.implicits.GetOps
+import io.circe.JsonObject
+import java.time.Instant
+import java.util.UUID
+import io.circe.generic.auto._, io.circe.parser._
+import cats.effect.Sync
+import cats.syntax.all._
+
+object Recover {
```

Comment:
Don't we want a `main()` to be able to recover already existing bad rows with the percentage issue ?

Reply:
what do you mean by ?

It’s added in Repeater.scala to main for the repeater
https://github.com/snowplow-incubator/snowplow-bigquery-loader/pull/139/files#diff-d0e5513b7035a18d3239b7e5b227c63eb1db4386df406d89c23b3b6e7e331cb5

Do you have something else in mind?

Reply:
I thought that there was 2 use cases:
1) auto recover failed inserts with the percentage issue
2) recover already existing bad rows

The `main()` in `Repeater.scala`Repeater.scala  does the first one but I struggle to see where 2) happens.

Reply:
This job is only to for point 2). The main starts the Recover stream, which does the job. Do I miss something?

Reply:
My understanding is that Repeater is a long running Scala app that reads from PubSub and writes to BQ. How will you make it run as a "batch" run to just read GCS ?

Reply:
As I see it recover in repeater should only be called for current failed inserts, not for the whole history.

Reply:
I added stream to Repeater that will work next to the main flow. It will read the data from GCS and write to Pub/Sub. Kind of similar to what repeater does, but the other way round. It will read all the data that are in filtered buckets, recover them, write to pub/sub and that's it. Stream will stop. All other streams with the main flow will continue to work normally. I'm only adding functionality, not changing the existing one. 

It's here: https://github.com/snowplow-incubator/snowplow-bigquery-loader/pull/139/files#diff-d0e5513b7035a18d3239b7e5b227c63eb1db4386df406d89c23b3b6e7e331cb5R55

Reply:
It means that we will start the repeater and will expect it to go and try to recover failed inserts from weeks ago. I guess what is troubling me is that we want to use a long-running streaming app to perform a batch job (for this very use case).

Reply:
That's why it's a one off job. There is nothing wrong in this solution, imho. We just add recovery feature to the app to not spin the new app and bother with all the infrastructure code and troubles spinning up something new. The long-term goal would be to use recovery project for this purpose, but currently it can not recover events in this part of the pipeline.